src/views/Home.jsx

```diff
@@ -1,11 +1,34 @@
 import './Home.css';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createNewList } from '../api/firebase';
+
+/** Home component that redirects a user to the List view if there is already a list created.
+ * If the user doesn't already have a list, a user can create one to be saved to Firestore and be redirected to the List view. */
+
+export function Home({ listToken, setListToken }) {
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+		}
+	}, []);
+
+	const createNewToken = async () => {
+		const newToken = generateToken();
+		setListToken(newToken);
+		createNewList(newToken);
+		navigate('/list');
+	};
 
```

Comment:
If we're importing the `createNewList()` async function from `firebase.js`, does it mean that we'll have to make the `createNewToken` function in the `Home.jsx` component async like we did here since it's using an async function within it, or does it not matter and we could have assigned `createNewToken` as a regular synchronous function?

Reply:
Great question! Since `createNewList` will need to be awaited in this function, `createNewToken` as a whole will be `async`.